kotlin_money/src/main/java/com/github/paylike/kotlin_money/PaymentAmount.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.paylike.kotlin_money
+
+import com.github.paylike.kotlin_currencies.PaylikeCurrency
+
+
+class PaymentAmount(_currency: PaylikeCurrency, _value: Long, _exponent: Int) {
+    val currency: PaylikeCurrency = _currency
+    val value: Long = _value
+    val exponent: Int = _exponent
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+        other as PaymentAmount
+        if (this.currency.code != other.currency.code) return false
+        return this.value == other.value && this.exponent == other.exponent
```

Comment:
This is not true in all logical cases, we should put a comment to the function indicating this.
Example:
```
{ currency: 'EUR', value: 150, exponent: 1 }
// will not be equal with
{ currency: 'EUR', value: 15, exponent: 0 }
```
Despite being equal in value. Which is fine for now I think, but makes sense to indicate this in a comment

Reply:
I see, I added a describing comment.